esmvaltool/utils/cmorizers/obs/osi_common.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import os
+
+import iris
+from iris.coord_categorisation import add_month, add_year
+from iris.analysis import MEAN
+
+from .utilities import (_set_global_atts, _convert_timeunits, _fix_coords,
+                        _fix_var_metadata, _save_variable)
+
+logger = logging.getLogger(__name__)
+
+
+def cmorize_osi(in_dir, out_dir, cfg,  hemisphere):
+    cmor_table = cfg['cmor_table']
+    glob_attrs = cfg['attributes']
+
+    logger.info("Starting cmorization for Tier%s OBS files: %s",
+                glob_attrs['tier'], glob_attrs['dataset_id'])
+    logger.info("Input data from: %s", in_dir)
+    logger.info("Output will be written to: %s", out_dir)
+
+    # run the cmorization
+    for var, vals in cfg['variables'].items():
+        for year in os.listdir(in_dir):
+            logger.info("CMORizing var %s for year %s", var, year)
+            var_info = cmor_table.get_variable(vals['mip'], var)
+            raw_info = {
+                'name': vals['raw'],
+                'file': os.path.join(
+                    in_dir, year, '??',
+                    '{0}_{1}_*.nc'.format(vals['raw'], hemisphere)
+                )
+            }
+            glob_attrs['mip'] = vals['mip']
+            extract_variable(var_info, raw_info, out_dir, glob_attrs, year)
+
+
+def extract_variable(var_info, raw_info, out_dir, attrs, year):
+    """Extract to all vars."""
+    var = var_info.short_name
+    cubes = iris.load(
+        raw_info['file'],
+        iris.Constraint(cube_func=lambda c: c.var_name == raw_info['name'])
+    )
+
+    tracking_ids = []
+    for cube in cubes:
+        tracking_ids.append(cube.attributes['tracking_id'])
+        to_remove = [
+            'time_coverage_start', 'time_coverage_end',
+            'history', 'tracking_id'
+        ]
+        for attr in to_remove:
+            del cube.attributes[attr]
+    cube = cubes.concatenate_cube()
```

Comment:
@jvegasbsc - I didn't realize you are the OP for this PR, sorry for including that link, I know you know the cases, I thought there was someone else authoring this PR and that might have been useful to them (obv not to you, Mr FixCubes :grinning: )

Reply:
We are only concatenating cubes that belongs to the same dataset, generated by the same procedure. 

I have not tested yet with the full dataset, but chances are that if we have same issues with other attributes it will be because of an issue with the data. I am quite happy with letting this case fail and force us to further investigate the issue